MAtrix/Matrix.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace C3L2
+{
+    public class Matrix <T>: IMatrix<T>
+    {
+
+        private int xCount;
+        private int yCount;
+        private T[,] body;
+        
+        public  Matrix(int xNum, int yNum)
+        {
+            xCount = xNum;
+            yCount = yNum;
+            body =  new T[xCount, yCount];
+        }
+
+        public int GetCount()
+        {
+            if (xCount == 0)
+                return yCount;
+            if (yCount == 0)
+                return xCount;
+            return xCount*yCount;
+        }
+
+        public T GetItem(int x, int y)
+        {
+            try
+            {
+                return body[x, y];
+            }
+            catch (ArgumentException ex)
```

Comment:
if you are rethrowing exception do not do this like "throw ex", use "throw" instead. As you are rethrowing all the exceptions, I see no point in catching them